test/src/org/omegat/util/LanguageTest.java

```diff
@@ -49,8 +51,9 @@ public void testGetLanguage() {
         String lang1 = "xx-YY";
         assertEquals(lang1, new Language(lang1).getLanguage());
 
-        String lang2 = "XX-yy";
-        assertEquals(lang2, new Language(lang2).getLanguage());
+        // Obsolete, the language (Locale) is always stored as lc-UC
+        // String lang2 = "XX-yy";
+        // assertEquals(lang2, new Language(lang2).getLanguage());
```

Comment:
As I understand it from bug #185, it's because external tools were picky about the case. 
It shouldn't be a problem internally since the Language class uses a valid Locale behind the scene. As the comparisons are done against the Locale.getLanguageCode(), the case should always be the same.
 
By the way, since `equals()` is overriden in this class, shouldn't we also override `hashCode()` ?

Reply:
> It shouldn't be a problem internally

When comparing `Language` to `Language`, yes, but I'm worried about when we have a locale string and compare it to `Language.toString()` or `Language.getLanguageCode()`. I tried to see if that happens anywhere within OmegaT and couldn't find anything, but a more thorough look seems prudent.

> By the way, since `equals()` is overriden in this class, shouldn't we also override `hashCode()` ?

Yes! I hope we don't use `Language` as a hashmap key anywhere.

Reply:
Good find on [bug #125](https://sourceforge.net/p/omegat/bugs/185/).

I think I'm OK with this now; could you add a comment like

```
// Previously, input case was intentionally preserved; see https://sourceforge.net/p/omegat/bugs/185/
```

> This is confusing, currently the equals uses `getLocaleCode` (which processes the Locale with some additional stuff) when I feel it would be safer to use `getLanguageCode` (directly from the Locale object).

The extra processing looks to be just swapping obsolete language identifiers. `Locale.toLanguageTag()` apparently does the same thing, so we could use that. But what about simply delegating `equals()` and `hashCode()` to `Locale`?

Reply:
Good idea, Language is really a thin wrapper around Locale at this point. Even the `languageCode` and `countryCode` fields are not very useful, as they're only used to initialize the Locale.